pkg/compactor/compactor_test.go

```diff
@@ -1656,6 +1656,9 @@ func prepareConfig(t *testing.T) Config {
 	compactorCfg.ShardingRing.Common.KVStore.Mock = inmem
 	compactorCfg.ShardingRing.Common.InstanceAddr = "localhost"
 
+	// The new default is 25m, but tests rely on the previous value of 0s
+	compactorCfg.CompactionWaitPeriod = 0
```

Comment:
I didn't dive into the details of this test, not sure if we should make it work with the new default instead.

Reply:
I don't think it affects the compaction algorithm itself so I think this is ok. I assume the wait period is tested separately ? Anyway, should be out of scope for this PR

Reply:
It's fine to set it to 0 here.